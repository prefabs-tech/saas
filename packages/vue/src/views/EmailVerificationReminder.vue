<template>
  <Page
    :title="t('user.emailVerification.title')"
    class="auth verify-email-reminder"
  >
    <Card>
      <p>{{ t("user.emailVerification.messages.verifyEmail") }}</p>

      <template #footer>
        <ButtonElement
          :label="t('user.emailVerification.button.label')"
          @click="handleResend"
        />
      </template>
    </Card>
  </Page>
</template>

<script lang="ts">
export default {
  name: "EmailVerificationReminder",
};
</script>

<script setup lang="ts">
import { useI18n } from "@dzangolab/vue3-i18n";
import { ButtonElement, Card, Page } from "@dzangolab/vue3-ui";

import { EMAIL_VERIFICATION } from "../constant";
import { useTranslations, emitter } from "../index";
import { resendVerificationEmail } from "../supertokens";

const messages = useTranslations();

const { t } = useI18n({ messages });

const handleResend = () => {
  resendVerificationEmail()
    .then((status) => {
      if (status === EMAIL_VERIFICATION.OK) {
        emitter.emit("notify", {
          text: t("user.emailVerification.messages.resend.success"),
          type: "success",
        });
      } else if (status === EMAIL_VERIFICATION.EMAIL_ALREADY_VERIFIED_ERROR) {
        emitter.emit("notify", {
          text: t("user.emailVerification.messages.resend.alreadyVerified"),
          type: "info",
        });
      }
    })
    .catch(() => {
      emitter.emit("notify", {
        text: t("user.emailVerification.messages.resend.error"),
        type: "error",
      });
    });
};
</script>
