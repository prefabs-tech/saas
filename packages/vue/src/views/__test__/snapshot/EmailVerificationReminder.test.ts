import i18Plugin, { useLocaleStore } from "@dzangolab/vue3-i18n";
import { mount} from "@vue/test-utils";
import { describe, it, expect } from "vitest";

import appConfig from "../../../components/__test__/config";
import EmailVerificationReminder from "../../EmailVerificationReminder.vue";

import type { VueWrapper } from "@vue/test-utils";

describe("EmailVerificationReminder", () => {
  const { setLocale } = useLocaleStore(appConfig.slug);

  const locales = ["en", "fr"];

  for (const locale of locales) {
    setLocale(locale);

    const wrapper: VueWrapper = mount(EmailVerificationReminder, {
      global: {
        plugins: [
          [
            i18Plugin,
            {
              config: appConfig,
            },
          ],
        ],
      },
    });

    it("matches snapshot in " + locale, () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  }
});
