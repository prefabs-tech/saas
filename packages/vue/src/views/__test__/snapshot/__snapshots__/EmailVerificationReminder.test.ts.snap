// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EmailVerificationReminder > matches snapshot in en 1`] = `
<div
  class="page auth verify-email-reminder"
  data-centered="false"
>
  <div
    class="page-header"
  >
    <div
      class="page-title-wrapper"
    >
      <div
        class="title"
      >
        <h1>
          Email verification
        </h1>
        
        <!---->
        
      </div>
      
      <!---->
      
    </div>
    <div
      class="page-toolbar"
    >
      
      
    </div>
  </div>
  <div
    class="page-content"
  >
    
    <div
      class="card"
    >
      <!---->
      <div
        class="card-body"
      >
        
        <p>
          Please check your email for verification
        </p>
        
      </div>
      <footer
        class="card-footer"
      >
        
        <button
          aria-label="button"
          class="button primary medium filled"
          role="button"
        >
          <!---->
          <div
            class="label"
          >
            
            <span>
              Resend email
            </span>
            
          </div>
          <!---->
          <!---->
        </button>
        
      </footer>
    </div>
    
    <!---->
  </div>
</div>
`;

exports[`EmailVerificationReminder > matches snapshot in fr 1`] = `
<div
  class="page auth verify-email-reminder"
  data-centered="false"
>
  <div
    class="page-header"
  >
    <div
      class="page-title-wrapper"
    >
      <div
        class="title"
      >
        <h1>
          Email verification
        </h1>
        
        <!---->
        
      </div>
      
      <!---->
      
    </div>
    <div
      class="page-toolbar"
    >
      
      
    </div>
  </div>
  <div
    class="page-content"
  >
    
    <div
      class="card"
    >
      <!---->
      <div
        class="card-body"
      >
        
        <p>
          Please check your email for verification
        </p>
        
      </div>
      <footer
        class="card-footer"
      >
        
        <button
          aria-label="button"
          class="button primary medium filled"
          role="button"
        >
          <!---->
          <div
            class="label"
          >
            
            <span>
              Resend email
            </span>
            
          </div>
          <!---->
          <!---->
        </button>
        
      </footer>
    </div>
    
    <!---->
  </div>
</div>
`;
